### PR TITLE
MOSIP-14678 RegProc: Packet error occurred at SUPERVISOR_VALIDATION stage for Active Supervisor.

### PR DESCRIPTION
**Pull request by neosoftSatish**

### CODE REVIEW

File: registration-processor/pre-processor/registration-processor-operator-validator-stage/src/main/java/io/mosip/registration/processor/stages/operatorvalidator/OperatorValidator.java [hunk 1]
```diff
@@ -362,9 +362,9 @@ private boolean validateMapping(List<String> pathsegments, InternalRegistrationS
 		regProcLogger.debug("validateMapping call ended for registrationId {} with response data {}",
 				registrationStatusDto.getRegistrationId(), JsonUtil.objectMapperObjectToJson(userDto));
 		if (userDto != null) {
-			if (responseWrapper.getErrors() == null) {
-				isValidUser = userDto.getRegistrationCenters().get(0).getIsActive();
-			} else {
+			isValidUser = userDto.getRegistrationCenters().get(0).getIsActive();
+		} else {
+			if (responseWrapper.getErrors() != null) {
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-operator-validator-stage/src/main/java/io/mosip/registration/processor/stages/operatorvalidator/OperatorValidator.java]: can we rather first check if errors are there, if yes return mapping not found, in else if check userDTO != null and in final else log a waring that error is not there and userDTO is empty and return false in that case ?

Replicate the same for supervisor also

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/pre-processor/registration-processor-operator-validator-stage/src/main/java/io/mosip/registration/processor/stages/operatorvalidator/OperatorValidator.java (hunk 1 above)
Comment (neosoftSatish) [reply registration-processor/pre-processor/registration-processor-operator-validator-stage/src/main/java/io/mosip/registration/processor/stages/operatorvalidator/OperatorValidator.java]: reconstructed this if else condition...

File: registration-processor/pre-processor/registration-processor-operator-validator-stage/src/main/java/io/mosip/registration/processor/stages/operatorvalidator/OperatorValidator.java [hunk 2]
```diff
@@ -361,16 +361,19 @@ private boolean validateMapping(List<String> pathsegments, InternalRegistrationS
...
+			List<ErrorDTO> error = responseWrapper.getErrors();
+			regProcLogger.debug("validateMapping call ended for registrationId {} with error data {}",
+					registrationStatusDto.getRegistrationId(), error.get(0).getMessage());
+			throw new BaseCheckedException(error.get(0).getMessage(),
+					StatusUtil.CENTER_DEVICE_MAPPING_NOT_FOUND.getCode());
+		} else if (userDto != null) {
+			isValidUser = userDto.getRegistrationCenters().get(0).getIsActive();
+		} else {
+			regProcLogger
+					.debug("validateMapping call ended for registrationId with error not there and userDTO is empty");
```
Comment (vishwa-vyom) [inline registration-processor/pre-processor/registration-processor-operator-validator-stage/src/main/java/io/mosip/registration/processor/stages/operatorvalidator/OperatorValidator.java]: Make this as warning log and change the message to say that "validateMapping call ended with no erros and userDTO as null so considering as mapping not valid"

Please take care of the line formatting

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: registration-processor/pre-processor/registration-processor-operator-validator-stage/src/main/java/io/mosip/registration/processor/stages/operatorvalidator/OperatorValidator.java (hunk 2 above)
Comment (neosoftSatish) [reply registration-processor/pre-processor/registration-processor-operator-validator-stage/src/main/java/io/mosip/registration/processor/stages/operatorvalidator/OperatorValidator.java]: changed the log message.

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**